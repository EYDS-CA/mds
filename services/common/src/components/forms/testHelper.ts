export const inputMeta = {
    touched: false,
    error: false,
    warning: false,
    autofilled: false,
    asyncValidating: false,
    dirty: false,
    dispatch: jest.fn(),
    form: "FORM",
    initial: undefined,
    invalid: false,
    pristine: true,
    submitFailed: false,
    submitting: false,
    valid: true,
    visited: false,
};

export const errorInputMeta = {
    touched: true,
    error: "Error message",
    warning: false,
    autofilled: false,
    asyncValidating: false,
    dirty: true,
    dispatch: jest.fn(),
    form: "FORM",
    initial: undefined,
    invalid: true,
    pristine: true,
    submitFailed: false,
    submitting: false,
    valid: true,
    visited: true,
}

// needs a value and a name
export const inputProps = {
    onBlur: jest.fn(),
    onChange: jest.fn(),
    onDragStart: jest.fn(),
    onDrop: jest.fn(),
    onFocus: jest.fn()
}