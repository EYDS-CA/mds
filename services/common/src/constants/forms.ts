export enum FORM {
  // projects
  ADD_EDIT_PROJECT_SUMMARY = "ADD_EDIT_PROJECT_SUMMARY",
  ADD_MINE_MAJOR_APPLICATION = "ADD_MINE_MAJOR_APPLICATION",
  // reports
  VIEW_EDIT_REPORT = "VIEW_EDIT_REPORT",
  // permits
  REVOKE_DIGITAL_CREDENTIAL = "REVOKE_DIGITAL_CREDENTIAL",
  REQUEST_REPORT = "REQUEST_REPORT",
  ADD_REPORT_TO_PERMIT_CONDITION = "ADD_REPORT_TO_PERMIT_CONDITION",
  ADD_PERMIT_CONDITION_CATEGORY = "ADD_PERMIT_CONDITION_CATEGORY",
  EDIT_PERMIT_CONDITION = "EDIT_PERMIT_CONDITION",
  // tailings
  ADD_TAILINGS = "ADD_TAILINGS",
  // documents
  REPLACE_DOCUMENT = "REPLACE_DOCUMENT",
  DELETE_DOCUMENT = "DELETE_DOCUMENT",
  ARCHIVE_DOCUMENT = "ARCHIVE_DOCUMENT",
  EDIT_HELP_GUIDE = "EDIT_HELP_GUIDE",
  INLINE_EDIT_PERMIT_CONDITION_CATEGORY = "INLINE_EDIT_PERMIT_CONDITION_CATEGORY",
  PERMIT_CONDITION_REVIEW_ASSIGNMENT = "PERMIT_CONDITION_REVIEW_ASSIGNMENT",
  UPDATE_MAJOR_MINE_APPLICATION = "UPDATE_MAJOR_MINE_APPLICATION",
  UPDATE_IRT = "UPDATE_IRT"
}
