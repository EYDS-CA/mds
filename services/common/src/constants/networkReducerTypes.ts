// Misc
export enum NetworkReducerTypes {
    LOAD_ALL_STATIC_CONTENT = "LOAD_ALL_STATIC_CONTENT",

    // Mines
    CREATE_MINE_RECORD = "CREATE_MINE_RECORD",
    CREATE_MINE_TYPE = "CREATE_MINE_TYPE",
    GET_MINE_RECORDS = "GET_MINE_RECORDS",
    GET_MINE_RECORD = "GET_MINE_RECORD",
    GET_SUBSCRIBED_MINES = "GET_SUBSCRIBED_MINES",
    GET_MINE_NAME_LIST = "GET_MINE_NAME_LIST",
    GET_MINE_BASIC_INFO_LIST = "GET_MINE_BASIC_INFO_LIST",
    GET_REGION_OPTIONS = "GET_REGION_OPTIONS",
    UPDATE_MINE_RECORD = "UPDATE_MINE_RECORD",
    CREATE_TSF = "CREATE_TSF",
    UPDATE_TSF = "UPDATE_TSF",
    GET_TSF = "GET_TSF",
    GET_MINE_DOCUMENTS = "GET_MINE_DOCUMENTS",
    ARCHIVE_MINE_DOCUMENTS = "ARCHIVE_MINE_DOCUMENTS",
    GET_MINE_VERIFIED_STATUS = "GET_MINE_VERIFIED_STATUS",
    SET_MINE_VERIFIED_STATUS = "SET_MINE_VERIFIED_STATUS",
    POLL_DOCUMENT_UPLOAD_STATUS = "POLL_DOCUMENT_UPLOAD_STATUS",

    // Mine Details
    REMOVE_MINE_TYPE = "REMOVE_MINE_TYPE",

    // Parties
    CREATE_PARTY = "CREATE_PARTY",
    UPDATE_PARTY = "UPDATE_PARTY",
    GET_PARTIES = "GET_PARTIES",
    GET_PARTY = "GET_PARTY",
    DELETE_PARTY = "DELETE_PARTY",
    GET_INSPECTORS = "GET_INSPECTORS",
    GET_PROJECT_LEADS = "GET_PROJECT_LEADS",
    ADD_PARTY_RELATIONSHIP = "ADD_PARTY_RELATIONSHIP",
    FETCH_PARTY_RELATIONSHIPS = "FETCH_PARTY_RELATIONSHIPS",
    REMOVE_PARTY_RELATIONSHIP = "REMOVE_PARTY_RELATIONSHIP",
    UPDATE_PARTY_RELATIONSHIP = "UPDATE_PARTY_RELATIONSHIP",
    ADD_DOCUMENT_TO_RELATIONSHIP = "ADD_DOCUMENT_TO_RELATIONSHIP",
    PARTY_ORGBOOK_ENTITY = "PARTY_ORGBOOK_ENTITY",
    MERGE_PARTIES = "MERGE_PARTIES",

    // OrgBook
    ORGBOOK_SEARCH = "ORGBOOK_SEARCH",
    ORGBOOK_CREDENTIAL = "ORGBOOK_CREDENTIAL",
    ORGBOOK_VERIFY = "ORGBOOK_VERIFY",

    // Compliance Orders
    GET_MINE_COMPLIANCE_INFO = "GET_MINE_COMPLIANCE_INFO",

    // MineSpace
    GET_MINESPACE_USER = "GET_MINESPACE_USER",
    CREATE_MINESPACE_USER = "CREATE_MINESPACE_USER",
    DELETE_MINESPACE_USER = "DELETE_MINESPACE_USER",
    GET_MINESPACE_USER_MINES = "GET_MINESPACE_USER_MINES",
    AUTHENTICATE_USER = "AUTHENTICATE_USER",
    UPDATE_MINESPACE_USER_MINES = "UPDATE_MINESPACE_USER_MINES",

    // MineSpace MCM contacts
    GET_MINISTRY_CONTACTS = "GET_MINISTRY_CONTACTS",
    UPDATE_MINISTRY_CONTACT = "UPDATE_MINISTRY_CONTACT",
    DELETE_MINISTRY_CONTACT = "DELETE_MINISTRY_CONTACT",

    // Permits
    CREATE_PERMIT = "CREATE_PERMIT",
    GET_PERMITS = "GET_PERMITS",
    UPDATE_PERMIT = "UPDATE_PERMIT",
    PATCH_PERMIT = "PATCH_PERMIT",
    CREATE_PERMIT_AMENDMENT = "CREATE_PERMIT_AMENDMENT",
    UPDATE_PERMIT_AMENDMENT = "UPDATE_PERMIT_AMENDMENT",
    GET_PERMIT_AMENDMENT = "GET_PERMIT_AMENDMENT",
    PERMIT_AMENDMENT_VC = "PERMIT_AMENDMENT_VC",
    UPDATE_PERMIT_AMENDMENT_DOCUMENT = "UPDATE_PERMIT_AMENDMENT_DOCUMENT",
    DELETE_PERMIT = "DELETE_PERMIT",
    DELETE_PERMIT_AMENDMENT = "DELETE_PERMIT_AMENDMENT",
    GET_PERMIT_CONDITIONS = "GET_PERMIT_CONDITIONS",
    CREATE_PERMIT_CONDITION = "CREATE_PERMIT_CONDITION",
    UPDATE_PERMIT_CONDITION = "UPDATE_PERMIT_CONDITION",
    DELETE_PERMIT_CONDITION = "DELETE_PERMIT_CONDITION",
    CREATE_PERMIT_CONDITION_CATEGORY = "CREATE_PERMIT_CONDITION_CATEGORY",
    UPDATE_PERMIT_CONDITION_CATEGORY = "UPDATE_PERMIT_CONDITION_CATEGORY",
    GET_PERMIT_CONDITION_CATEGORIES = "GET_PERMIT_CONDITION_CATEGORIES",
    DELETE_PERMIT_CONDITION_CATEGORY = "DELETE_PERMIT_CONDITION_CATEGORY",

    // Explosive Storage & Use Permits
    CREATE_EXPLOSIVES_PERMIT = "CREATE_EXPLOSIVES_PERMIT",
    GET_EXPLOSIVES_PERMITS = "GET_EXPLOSIVES_PERMITS",
    UPDATE_EXPLOSIVES_PERMIT = "UPDATE_EXPLOSIVES_PERMIT",
    DELETE_EXPLOSIVES_PERMIT = "DELETE_EXPLOSIVES_PERMIT",
    CREATE_EXPLOSIVES_PERMIT_AMENDMENT = "CREATE_EXPLOSIVES_PERMIT_AMENDMENT",
    UPDATE_EXPLOSIVES_PERMIT_AMENDMENT = "UPDATE_EXPLOSIVES_PERMIT_AMENDMENT",

    // Search
    GET_SEARCH_OPTIONS = "GET_SEARCH_OPTIONS",
    GET_SEARCH_RESULTS = "GET_SEARCH_RESULTS",
    GET_SEARCH_BAR_RESULTS = "GET_SEARCH_BAR_RESULTS",
    CLEAR_SEARCH_BAR_RESULTS = "CLEAR_SEARCH_BAR_RESULTS",
    CLEAR_ALL_SEARCH_RESULTS = "CLEAR_ALL_SEARCH_RESULTS",

    // Mine Subscriptions
    SUBSCRIBE = "SUBSCRIBE",
    UNSUBSCRIBE = "UNSUBSCRIBE",

    // Variances
    CREATE_MINE_VARIANCE = "CREATE_MINE_VARIANCE",
    GET_VARIANCES = "GET_VARIANCES",
    GET_VARIANCE = "GET_VARIANCE",
    ADD_DOCUMENT_TO_VARIANCE = "ADD_DOCUMENT_TO_VARIANCE",
    REMOVE_DOCUMENT_FROM_VARIANCE = "REMOVE_DOCUMENT_FROM_VARIANCE",
    UPDATE_MINE_VARIANCE = "UPDATE_MINE_VARIANCE",
    DELETE_VARIANCE = "DELETE_VARIANCE",

    // Projects
    UPDATE_PROJECT = "UPDATE_PROJECT",
    GET_PROJECTS = "GET_PROJECTS",
    GET_PROJECT = "GET_PROJECT",
    CREATE_MINE_PROJECT_SUMMARY = "CREATE_MINE_PROJECT_SUMMARY",
    GET_PROJECT_SUMMARIES = "GET_PROJECT_SUMMARIES",
    GET_PROJECT_SUMMARY = "GET_PROJECT_SUMMARY",
    GET_PROJECT_SUMMARY_MINISTRY_COMMENTS = "GET_PROJECT_SUMMARY_MINISTRY_COMMENTS",
    CREATE_PROJECT_SUMMARY_MINISTRY_COMMENTS = "CREATE_PROJECT_SUMMARY_MINISTRY_COMMENTS",
    REMOVE_DOCUMENT_FROM_PROJECT_SUMMARY = "REMOVE_DOCUMENT_FROM_PROJECT_SUMMARY",
    UPDATE_MINE_PROJECT_SUMMARY = "UPDATE_MINE_PROJECT_SUMMARY",
    DELETE_PROJECT_SUMMARY = "DELETE_PROJECT_SUMMARY",
    POST_NEW_DOCUMENT_VERSION = "POST_NEW_DOCUMENT_VERSION",
    INFORMATION_REQUIREMENTS_TABLE = "INFORMATION_REQUIREMENTS_TABLE",
    GET_REQUIREMENTS = "GET_REQUIREMENTS",
    GET_INFORMATION_REQUIREMENTS_TABLE = "GET_INFORMATION_REQUIREMENTS_TABLE",
    UPDATE_INFORMATION_REQUIREMENTS_TABLE = "UPDATE_INFORMATION_REQUIREMENTS_TABLE",
    REMOVE_DOCUMENT_FROM_INFORMATION_REQUIREMENTS_TABLE =
    "REMOVE_DOCUMENT_FROM_INFORMATION_REQUIREMENTS_TABLE",
    CREATE_MAJOR_MINES_APPLICATION = "CREATE_MAJOR_MINE_APLICATION",
    GET_MAJOR_MINES_APPLICATION = "GET_MAJOR_MINES_APPLICATION",
    UPDATE_MAJOR_MINES_APPLICATION = "UPDATE_MAJOR_MINE_APPLICATION",
    REMOVE_DOCUMENT_FROM_MAJOR_MINE_APPLICATION =
    "REMOVE_DOCUMENT_FROM_MAJOR_MINE_APPLICATION",
    CREATE_PROJECT_DECISION_PACKAGE = "CREATE_PROJECT_DECISION_PACKAGE",
    UPDATE_PROJECT_DECISION_PACKAGE = "UPDATE_PROJECT_DECISION_PACKAGE",
    REMOVE_DOCUMENT_FROM_PROJECT_DECISION_PACKAGE =
    "REMOVE_DOCUMENT_FROM_PROJECT_DECISION_PACKAGE",
    DOCUMENTS_COMPRESSION = "DOCUMENTS_COMPRESSION",
    POLL_DOCUMENTS_COMPRESSION_PROGRESS = "POLL_DOCUMENTS_COMPRESSION_PROGRESS",
    CREATE_PROJECT_LINKS = "CREATE_PROJECT_LINKS",
    DELETE_PROJECT_LINK = "DELETE_PROJECT_LINK",

    // Incidents
    CREATE_MINE_INCIDENT = "CREATE_MINE_INCIDENT",
    GET_INCIDENTS = "GET_INCIDENTS",
    GET_MINE_INCIDENTS = "GET_MINE_INCIDENTS",
    GET_MINE_INCIDENT = "GET_MINE_INCIDENT",
    UPDATE_MINE_INCIDENT = "UPDATE_MINE_INCIDENT",
    REMOVE_DOCUMENT_FROM_MINE_INCIDENT = "REMOVE_DOCUMENT_FROM_MINE_INCIDENT",
    DELETE_MINE_INCIDENT = "DELETE_MINE_INCIDENT",
    GET_MINE_INCIDENT_NOTES = "GET_MINE_INCIDENT_NOTES",
    CREATE_MINE_INCIDENT_NOTE = "CREATE_MINE_INCIDENT_NOTE",
    DELETE_MINE_INCIDENT_NOTE = "DELETE_MINE_INCIDENT_NOTE",

    // Work Information
    CREATE_MINE_WORK_INFORMATION = "CREATE_MINE_WORK_INFORMATION",
    GET_MINE_WORK_INFORMATIONS = "GET_MINE_WORK_INFORMATIONS",
    UPDATE_MINE_WORK_INFORMATION = "UPDATE_MINE_WORK_INFORMATION",
    DELETE_MINE_WORK_INFORMATION = "DELETE_MINE_WORK_INFORMATION",

    // Reports
    GET_REPORTS = "GET_REPORTS",
    UPDATE_MINE_REPORT = "UPDATE_MINE_REPORT",
    CREATE_MINE_REPORT = "CREATE_MINE_REPORT",
    DELETE_MINE_REPORT = "DELETE_MINE_REPORT",
    GET_MINE_REPORTS = "GET_MINE_REPORTS",
    GET_MINE_REPORT = "GET_MINE_REPORT",

    // Report Comments
    GET_MINE_REPORT_COMMENTS = "GET_MINE_REPORT_COMMENTS",
    UPDATE_MINE_REPORT_COMMENT = "UPDATE_MINE_REPORT_COMMENT",
    CREATE_MINE_REPORT_COMMENT = "CREATE_MINE_REPORT_COMMENT",
    DELETE_MINE_REPORT_COMMENT = "DELETE_MINE_REPORT_COMMENT",

    // Notices of Work
    GET_NOTICE_OF_WORK_APPLICATIONS = "GET_NOTICE_OF_WORK_APPLICATIONS",
    GET_NOTICE_OF_WORK_APPLICATION = "GET_NOTICE_OF_WORK_APPLICATION",
    GET_MINE_NOTICE_OF_WORK_APPLICATIONS = "GET_MINE_NOTICE_OF_WORK_APPLICATIONS",
    IMPORT_NOTICE_OF_WORK_APPLICATION = "IMPORT_NOTICE_OF_WORK_APPLICATION",
    CREATE_NOTICE_OF_WORK_APPLICATION_IMPORT_SUBMISSION_DOCUMENTS_JOB =
    "CREATE_NOTICE_OF_WORK_APPLICATION_IMPORT_SUBMISSION_DOCUMENTS_JOB",
    CREATE_NOTICE_OF_WORK_APPLICATION = "CREATE_NOTICE_OF_WORK_APPLICATION",
    CREATE_ADMIN_AMENDMENT_APPLICATION = "CREATE_ADMIN_AMENDMENT_APPLICATION",
    UPDATE_NOTICE_OF_WORK_APPLICATION = "UPDATE_NOTICE_OF_WORK_APPLICATION",
    GET_IMPORTED_NOTICE_OF_WORK_APPLICATION = "GET_IMPORTED_NOTICE_OF_WORK_APPLICATION",
    GET_ORIGINAL_NOTICE_OF_WORK_APPLICATION = "GET_ORIGINAL_NOTICE_OF_WORK_APPLICATION",
    CREATE_NOTICE_OF_WORK_APPLICATION_PROGRESS =
    "CREATE_NOTICE_OF_WORK_APPLICATION_PROGRESS",
    UPDATE_NOTICE_OF_WORK_APPLICATION_PROGRESS =
    "UPDATE_NOTICE_OF_WORK_APPLICATION_PROGRESS",
    CREATE_NOTICE_OF_WORK_APPLICATION_REVIEW = "CREATE_NOTICE_OF_WORK_APPLICATION_REVIEW",
    GET_NOTICE_OF_WORK_APPLICATION_REVIEW = "GET_NOTICE_OF_WORK_APPLICATION_REVIEW",
    UPDATE_NOTICE_OF_WORK_APPLICATION_REVIEW = "UPDATE_NOTICE_OF_WORK_APPLICATION_REVIEW",
    REMOVE_NOTICE_OF_WORK_APPLICATION_REVIEW = "REMOVE_NOTICE_OF_WORK_APPLICATION_REVIEW",
    REMOVE_NOTICE_OF_WORK_APPLICATION_DOCUMENT =
    "REMOVE_NOTICE_OF_WORK_APPLICATION_DOCUMENT",
    ADD_DOCUMENT_TO_NOTICE_OF_WORK = "ADD_DOCUMENT_TO_NOTICE_OF_WORK",
    SORT_NOTICE_OF_WORK_DOCUMENTS = "SORT_NOTICE_OF_WORK_DOCUMENTS",
    GET_IMPORT_NOTICE_OF_WORK_SUBMISSION_DOCUMENTS_JOB =
    "GET_IMPORT_NOTICE_OF_WORK_SUBMISSION_DOCUMENTS_JOB",
    UPDATE_NOTICE_OF_WORK_STATUS = "UPDATE_NOTICE_OF_WORK_STATUS",
    CREATE_NOTICE_OF_WORK_APPLICATION_DELAY = "CREATE_NOTICE_OF_WORK_APPLICATION_DELAY",
    UPDATE_NOTICE_OF_WORK_APPLICATION_DELAY = "UPDATE_NOTICE_OF_WORK_APPLICATION_DELAY",
    FETCH_NOTICE_OF_WORK_APPLICATION_DELAY = "FETCH_NOTICE_OF_WORK_APPLICATION_DELAY",

    // Securities
    GET_MINE_BONDS = "GET_MINE_BONDS",
    CREATE_BOND = "CREATE_BOND",
    UPDATE_BOND = "UPDATE_BOND",
    TRANSFER_BOND = "TRANSFER_BOND",
    GET_MINE_RECLAMATION_INVOICES = "GET_MINE_RECLAMATION_INVOICES",
    CREATE_RECLAMATION_INVOICE = "CREATE_RECLAMATION_INVOICE",
    UPDATE_RECLAMATION_INVOICE = "UPDATE_RECLAMATION_INVOICE",

    // Comments
    GET_MINE_COMMENTS = "GET_MINE_COMMENTS",
    CREATE_MINE_COMMENTS = "CREATE_MINE_COMMENTS",
    DELETE_MINE_COMMENT = "DELETE_MINE_COMMENT",

    // EPIC Information
    GET_MINE_EPIC_INFO = "GET_MINE_EPIC_INFO",

    // Notices of Departure
    CREATE_NOTICE_OF_DEPARTURE = "CREATE_NOTICE_OF_DEPARTURE",
    UPDATE_NOTICE_OF_DEPARTURE = "UPDATE_NOTICE_OF_DEPARTURE",
    GET_NOTICES_OF_DEPARTURE = "GET_NOTICES_OF_DEPARTURE",
    ADD_DOCUMENT_TO_NOTICE_OF_DEPARTURE = "ADD_DOCUMENT_TO_NOTICE_OF_DEPARTURE",
    GET_DETAILED_NOTICE_OF_DEPARTURE = "GET_DETAILED_NOTICE_OF_DEPARTURE",

    // Activities
    GET_ACTIVITIES = "GET_ACTIVITIES",

    // Dams
    CREATE_DAM = "CREATE_DAM",
    UPDATE_DAM = "UPDATE_DAM",
    GET_DAM = "GET_DAM",

    //Verifiable Credentials
    CREATE_VC_WALLET_CONNECTION_INVITATION = "CREATE_VC_WALLET_CONNECTION_INVITATION",
    FETCH_VC_WALLET_CONNECTION_INVITATIONS = "FETCH_VC_WALLET_CONNECTION_INVITATIONS",
    ISSUE_VC = "ISSUE_VC",
    DELETE_VC_WALLET_CONNECTION = "DELETE_VC_WALLET_CONNECTION",

    // CORE-specific
    // Documents
    GENERATE_NOTICE_OF_WORK_APPLICATION_DOCUMENT = "GENERATE_NOTICE_OF_WORK_APPLICATION_DOCUMENT",
    GET_NOTICE_OF_WORK_APPLICATION_DOCUMENT_CONTEXT_TEMPLATE = "GET_NOTICE_OF_WORK_APPLICATION_DOCUMENT_CONTEXT_TEMPLATE",
    EXPORT_NOTICE_OF_WORK_APPLICATION_DOCUMENT = "EXPORT_NOTICE_OF_WORK_APPLICATION_DOCUMENT",
    GENERATE_EXPLOSIVES_PERMIT_DOCUMENT = "GENERATE_EXPLOSIVES_PERMIT_DOCUMENT",
    GET_EXPLOSIVES_PERMIT_DOCUMENT_CONTEXT_TEMPLATE = "GET_EXPLOSIVES_PERMIT_DOCUMENT_CONTEXT_TEMPLATE",
    // Alerts
    CREATE_MINE_ALERTS = "CREATE_MINE_ALERTS",
    UPDATE_MINE_ALERT = "UPDATE_MINE_ALERT",
    DELETE_MINE_ALERT = "DELETE_MINE_ALERT",
    GET_MINE_ALERTS = "GET_MINE_ALERTS",
    GET_GLOBAL_MINE_ALERTS = "GET_GLOBAL_MINE_ALERTS",

    // MS-specific
    GET_USER_INFO = "GET_USER_INFO",
    GET_USER_MINE_INFO = "GET_USER_MINE_INFO"
}