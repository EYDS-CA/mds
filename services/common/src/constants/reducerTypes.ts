// Misc
export const AUTHENTICATION = "AUTHENTICATION";
export const STATIC_CONTENT = "STATIC_CONTENT";
export const MODAL = "MODAL";
export const DOCUMENT_VIEWER = "DOCUMENT_VIEWER";
export const SYSTEM_FLAG = "SYSTEM_FLAG";

// Mines
export const MINES = "MINES";

// Parties
export const PARTIES = "PARTIES";

// OrgBook
export const ORGBOOK = "ORGBOOK";

// Compliance Orders
export const COMPLIANCE = "COMPLIANCE";

// MineSpace
export const MINESPACE = "MINESPACE";

// Permits
export const PERMITS = "PERMITS";

// Explosive Storage & Use Permits
export const EXPLOSIVES_PERMITS = "EXPLOSIVES_PERMITS";

// Search
export const SEARCH = "SEARCH";

// Variances
export const VARIANCES = "VARIANCES";

// Projects
export const PROJECTS = "PROJECTS";
export const PROJECT_SUMMARIES = "PROJECT_SUMMARIES";
export const INFORMATION_REQUIREMENTS_TABLE = "INFORMATION_REQUIREMENTS_TABLE";

// Core Users
export const USERS = "USERS";

// Incidents
export const INCIDENTS = "INCIDENTS";

// Work Information
export const MINE_WORK_INFORMATIONS = "MINE_WORK_INFORMATIONS";

// Reports
export const REPORTS = "REPORTS";

// Notices of Work
export const NOTICE_OF_WORK = "NOTICE_OF_WORK";

// Securities
export const SECURITIES = "SECURITIES";

// EPIC Information
export const EXTERNAL_AUTHS = "EXTERNAL_AUTHS";

// Notices of Departure
export const NOTICES_OF_DEPARTURE = "NOTICES_OF_DEPARTURE";

// Activities
export const ACTIVITIES = "ACTIVITIES";

// Tailings Storage Facilities
export const TAILINGS = "TAILINGS";

// Dams
export const DAMS = "DAMS";


//Verifiable Credentials
export const VERIFIABLE_CREDENTIALS = "VERIFIABLE_CREDENTIALS";
