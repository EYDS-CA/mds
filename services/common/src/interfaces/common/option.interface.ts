export interface IOption {
  label: string;
  value: string | number | boolean;
  tooltip?: string;
}

export interface IGroupedDropdownList {
  groupName: string | number;
  opt: IOption[];
}
