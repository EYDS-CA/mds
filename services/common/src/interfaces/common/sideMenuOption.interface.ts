import { ReactNode } from "react";

export interface ISideMenuOption {
  href: string;
  title: string | ReactNode;
  icon?: ReactNode;
  description?: string | ReactNode;
}
