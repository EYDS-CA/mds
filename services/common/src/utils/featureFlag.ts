import { detectProdEnvironment as IN_PROD } from "./environmentUtils";
import flagsmith from "flagsmith";

// Name of feature flags. These correspond to feature flags defined in flagsmith.
export enum Feature {
  MAJOR_PROJECT_ARCHIVE_FILE = "major_project_archive_file",
  DOCUMENTS_REPLACE_FILE = "major_project_replace_file",
  MAJOR_PROJECT_ALL_DOCUMENTS = "major_project_all_documents",
  MAJOR_PROJECT_DECISION_PACKAGE = "major_project_decision_package",
  FLAGSMITH = "flagsmith",
  TSF_V2 = "tsf_v2",
}

// Definition of legacy flags
// TODO: Remove after flagsmith is live
const Flags = {
  [Feature.MAJOR_PROJECT_ARCHIVE_FILE]: !IN_PROD(),
  [Feature.DOCUMENTS_REPLACE_FILE]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_ALL_DOCUMENTS]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_DECISION_PACKAGE]: !IN_PROD(),
  [Feature.TSF_V2]: !IN_PROD(),
  [Feature.FLAGSMITH]: !IN_PROD(),
};

export const initializeFlagsmith = async (flagsmithUrl, flagsmithKey) => {
  await flagsmith.init({
    api: flagsmithUrl,
    environmentID: flagsmithKey,
    cacheFlags: true,
    enableAnalytics: true,
  });
};

/**
 * Returns true if the given feature is enabled
 * @param feature Feature to verify
 * @returns true if the given feature is enabled
 */
export const isFeatureEnabled = (feature: Feature) => {
  if (Feature.FLAGSMITH in Flags && Flags[Feature.FLAGSMITH]) {
    return flagsmith.hasFeature(feature);
  }

  if (feature in Flags) {
    return Flags[feature];
  }

  return false;
};
