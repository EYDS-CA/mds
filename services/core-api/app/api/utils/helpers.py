import re


def clean_HTML_string(raw_html):
    cleanr = re.compile('<.*?>')
    cleantext = re.sub(cleanr, '', raw_html)
    return cleantext


"""
Add sufix to permit number adding one to versioning.
"""


def generate_permit_no_sufix(permit, separator, filling=2):
    result = permit.split(separator + "-")
    version_str = '0' if len(result) < 2 else result[1]
    version = str(int(version_str) + 1).zfill(filling)
    permit_no = separator + "-" + version

    return permit_no
