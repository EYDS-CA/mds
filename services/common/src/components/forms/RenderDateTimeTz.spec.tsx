import React from "react";
import { render } from "@testing-library/react";
import RenderDateTimeTz from "./RenderDateTimeTz";
import { inputMeta, inputProps } from "./testHelper";
import { normalizeDatetime } from "@mds/common/redux/utils/helpers";
import { ReduxWrapper } from "@mds/common/tests/utils/ReduxWrapper";
import { DEFAULT_TIMEZONE } from "@mds/common/constants";

const initialState = {
    form: {
        [inputMeta.form]: {
            values: { "tz_field_name": DEFAULT_TIMEZONE }
        }
    }
};
describe("RenderDateTimeTz", () => {
    it("renders properly", () => {
        const { container } = render(
            <ReduxWrapper initialState={initialState}>
                <RenderDateTimeTz
                    id="dateTzInput"
                    normalize={normalizeDatetime}
                    input={{ value: new Date("2025-01-16"), name: "dateTzInput", ...inputProps }}
                    meta={inputMeta}
                    timezoneFieldProps={{ name: "tz_field_name" }}
                /></ReduxWrapper>);
        expect(container).toMatchSnapshot();
    })
})