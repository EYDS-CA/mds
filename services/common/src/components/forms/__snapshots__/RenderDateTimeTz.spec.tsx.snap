// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderDateTimeTz renders properly 1`] = `
<div>
  <div
    class="ant-form-item ant-form-item-with-help"
  >
    <div
      class="ant-row ant-form-item-row"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-picker"
            >
              <div
                class="ant-picker-input"
              >
                <input
                  autocomplete="off"
                  id="dateTzInput"
                  placeholder="Select date"
                  readonly=""
                  size="29"
                  title="2025-01-15 16:00 -08:00 PST"
                  value="2025-01-15 16:00 -08:00 PST"
                />
                <span
                  class="ant-picker-suffix"
                >
                  <span
                    aria-label="calendar"
                    class="anticon anticon-calendar"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="calendar"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="ant-picker-clear"
                  role="button"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          style="display: flex; flex-wrap: nowrap;"
        >
          <div
            class="ant-form-item-explain ant-form-item-explain-connected"
            role="alert"
          >
            <div
              class=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
